src/shared/bpf-program.c

```diff
@@ -309,6 +309,13 @@ int bpf_map_new(enum bpf_map_type type, size_t key_size, size_t value_size, size
         attr.value_size = value_size;
         attr.max_entries = max_entries;
         attr.map_flags = flags;
+        strncpy(attr.map_name, name, sizeof(attr.map_name)-1);
+        for (int i = 0; attr.map_name[i]; i++) {
+                if (strchr("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_.",
```

Comment:
we have macros such as ALPHANUMERICAL that make this a lot more readable:

```c
if (strchr(ALPHANUMERICAL "_.", …
```

also, please add a comment here:

```c
/* as per https://www.kernel.org/doc/html/next/bpf/maps.html#usage-notes the allowed charset is alphanumerical + "." and "_" */